File: ruby/testcode/deserial_021.rb
require_relative 'shared'

# vuln-code-snippet start ruby_deser_msgpack_type
def msgpack_type_deserialize_handler(req)
  MessagePack::DefaultFactory.register_type(0x01, Object)
  obj = MessagePack.unpack(req.post('data'))  # vuln-code-snippet vuln-line ruby_deser_msgpack_type
  BenchmarkResponse.json({ result: obj.to_s })
end
# vuln-code-snippet end ruby_deser_msgpack_type

File: php/testcode/deserial_021.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_deser_gzuncompress_chain
function deserial021(BenchmarkRequest $req): BenchmarkResponse {
    $payload = $req->post('payload');
    $decompressed = gzuncompress(base64_decode($payload));
    $obj = unserialize($decompressed); // vuln-code-snippet vuln-line php_deser_gzuncompress_chain
    return BenchmarkResponse::ok('processed');
}
// vuln-code-snippet end php_deser_gzuncompress_chain
